integration/hive/src/main/java/org/apache/carbondata/hive/CarbonHiveRecordReader.java

```diff
@@ -88,17 +90,12 @@ private void initialize(InputSplit inputSplit, Configuration conf) throws IOExce
     } catch (QueryExecutionException e) {
       throw new IOException(e.getMessage(), e.getCause());
     }
-    if (valueObj == null) {
-      valueObj =
-          new ArrayWritable(Writable.class, new Writable[queryModel.getProjectionColumns().length]);
-    }
-
     final TypeInfo rowTypeInfo;
     final List<String> columnNames;
     List<TypeInfo> columnTypes;
     // Get column names and sort order
     final String colIds = conf.get("hive.io.file.readcolumn.ids");
-    final String columnNameProperty = conf.get("hive.io.file.readcolumn.names");
+    final String columnNameProperty = conf.get(serdeConstants.LIST_COLUMNS);
```

Comment:
why change to serdeConstants.LIST_COLUMNS？ I had got columnNameProperty from serdeConstants.LIST_COLUMNS, but I found that the columns in serdeConstants.LIST_COLUMNS  sometimes contains columns which don't belong to this table